django/thunderstore/api/cyberstorm/serializers/team.py

```diff
@@ -1,15 +1,45 @@
+from typing import Optional
+
 from rest_framework import serializers
 
-from .user import CyberstormUserSerializer
+
+class CyberstormTeamSerializer(serializers.Serializer):
+    """
+    This is for team's public profile and readably by anyone. Don't add
+    any sensitive information here.
+    """
+
+    identifier = serializers.IntegerField(source="id")
+    name = serializers.CharField()
+    donation_link = serializers.CharField(required=False)
 
 
 class CyberstormTeamMemberSerializer(serializers.Serializer):
-    user = CyberstormUserSerializer()
+    identifier = serializers.IntegerField(source="user.id")
+    username = serializers.CharField(source="user.username")
+    avatar = serializers.SerializerMethodField()
     role = serializers.CharField()
 
+    def get_avatar(self, obj) -> Optional[str]:
+        """
+        TODO: we might want to consider adding "aggregated fields" to
+              User and compute the return value in a background task.
+        """
+        for connection in obj.user.social_auth.all():
+            if connection.provider == "github" and connection.extra_data.get(
+                "avatar_url",
+            ):
+                return connection.extra_data.get("avatar_url")
 
-class CyberstormTeamSerializer(serializers.Serializer):
-    identifier = serializers.CharField(source="id")
-    name = serializers.CharField()
-    members = CyberstormTeamMemberSerializer(many=True)
-    donation_link = serializers.CharField(required=False)
+            if connection.provider == "overwolf" and connection.extra_data.get(
+                "avatar",
+            ):
+                return connection.extra_data.get("avatar")
+
+        return None
```

Comment:
Non-blocker:

Perhaps the avatar fetching logic is worth extracting to a function somewhere more fitting at least? Would make testing easy too.

Reply:
Done in a [separate commit](https://github.com/thunderstore-io/Thunderstore/pull/892/commits/5cd6dd1d3a41c17450b55eb5f09754cafea55df5).